spec/controllers/sign_up/passwords_controller_spec.rb

```diff
@@ -29,11 +26,8 @@
     end
 
     it 'tracks an invalid password event' do
-      user = create(:user, :unconfirmed)
-      token, = Devise.token_generator.generate(User, :confirmation_token)
-      user.update(
-        confirmation_token: token, confirmation_sent_at: Time.current
-      )
+      token = Devise.token_generator.generate(User, :confirmation_token)
```

Comment:
Thinking about this, I don't think we even need to invoke Devise at all. We just need to make sure the passed in token matches what's in the DB, so we can set it to an old String.